ionc/ion_binary.c

```diff
@@ -117,7 +117,7 @@ int ion_binary_len_ion_decimal(decQuad *value, decContext *context )
 int ion_binary_len_ion_float( double value )
 {
     int len = 0;
-    if (value == 0) {
+    if (value == 0 && !ion_float_is_negative_zero(value)) {
```

Comment:
len is already initialized to 0 -- there's no need to set it to 0 again in this if block.  Consider simplifying to:
```
int len = 0;
if (value != 0 || ion_float_is_negative_zero(value)) {
    len = sizeof(value);
}
return len;
```